app/models/tagging.rb

```diff
@@ -3,7 +3,6 @@ class Tagging < ApplicationRecord
   belongs_to :taggable, polymorphic: true, touch: true
 
   validates_presence_of :tagger
-  validates_associated :taggable
```

Comment:
This code didn't originally validate `taggable`. In the Rails 4.0 update, this changed from
```
validates_presence_of :tagger, :taggable
```
to
```
validates_presence_of :tagger
validates_associated :taggable
```

I'm guessing the reason the `validates_presence_of :taggable` check was disabled is because the inverses aren't set up properly on taggings, which meant that the validation always failed on new works. Though I'm not sure why the check was changed to `validates_associated` (instead of trying to fix the inverses or deleting it entirely).

The issue with `validates_associated` is that it will load a fresh copy of the taggable object from the database (because broken inverses) and try to validate it. But that fresh copy will be missing any info that isn't in the database when the `Tagging` is being saved. When saving the work for the first time, this includes almost all associations, because `Taggable` is the first module included in works, and I'm pretty sure associations are saved in the order that they're declared. In particular, the version of the work that it validates during work creation will always be missing all creatorship information, making it look like the work has no creators.

(It works on master because any validity checks involving the presence of associations are coded up as `before_save` hooks, instead of `validate` methods, so that they're not called when `validates_associated` tries to check it. But that causes its own issue: it won't display the errors until the user actually tries to save the work. So if you want those issues to be flagged when the user is previewing, you have to add code to the `WorksController` for each validation that's implemented as a `before_save` hook. For example, `WorksController#set_work_tag_error_messages` vs. `Taggable#validate_tags`. This results in the `WorksController` being more complex than it ought to be.)

The obvious solution to this is to fix up the inverses. Unfortunately, fixing the inverses is more complicated than it sounds because of [things like this](https://github.com/otwcode/otwarchive/pull/3326#issuecomment-394096066):
>Works avoid this problem because the before_destroy hook in the Tagging class runs remove_filter_tagging an additional time for each deleted tag. **Because the taggable association in Taggings isn't set up as the inverse of the taggings association in Works,** the call from the Tagging class loads a fresh copy of the work for every tagging deleted or added. So the first time remove_filter_tagging is run -- from the Tagging class -- **the tags association is loaded fresh, resulting in more SQL queries and processing time, but also the correct outcome.** The second time remove_filter_tagging is run -- from the Taggable class -- it's a no-op. And there's no equivalent of check_filter_taggings.

Basically: Using `validates_associated` while the inverses are broken means that validity of `taggable` may not be the same as the validity of the work whose tags are being edited. So I don't think it's a very useful check at the moment. But fixing the inverses will break the calculation of filters, and including the fix for that would make this PR even more unmanageable. In my opinion, the cleanest way to handle this is to delete the `validates_associated`.

Reply:
Thanks for explaining this! Leaving out this validation sounds good.